common/src/main/java/com/linecorp/centraldogma/common/AggregatedEntry.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.common;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.MoreObjects;
+
+/**
+ * Aggregated entry in a repository.
+ *
+ * @param <T> the content type. It is {@link JsonNode} because only JSON aggregation is currently supported.
+ */
+public final class AggregatedEntry<T> {
+
+    /**
+     * Returns a newly-created {@link AggregatedEntry}.
+     *
+     * @param type the type of the {@link Entry}
+     * @param content the content of the {@link Entry}
+     * @param <T> the content type. It is {@link JsonNode} because only JSON aggregation is currently supported.
+     */
+    public static <T> AggregatedEntry<T> of(EntryType type, T content) {
+        return new AggregatedEntry<>(type, content);
+    }
+
+    private final EntryType type;
+    private final T content;
+    @Nullable
+    private String contentAsText;
+    @Nullable
+    private String contentAsPrettyText;
+
+    /**
+     * Creates a new instance.
+     */
+    private AggregatedEntry(EntryType type, T content) {
+        this.type = requireNonNull(type, "type");
+        this.content = requireNonNull(content, "content");
```

Comment:
Check if `content` is instance of `type.type()`.

Reply:
Fixed. Thank you~!